vertx-web/src/test/java/io/vertx/ext/web/handler/RerouteTest.java

```diff
@@ -246,4 +246,12 @@ public void testRerouteChecksWithFragment() throws Exception {
 
     testRequest(HttpMethod.GET, "/base?p=1", 200, "OK", "/other");
   }
+
+  @Test
+  public void testRerouteWhenBaseRequestHasBadlyEncodedParams() throws Exception {
+    router.get("/other").handler(ctx -> ctx.response().end("/other"));
```

Comment:
Can we add an assert here to also check the params map contents after reroute? E.g.
```
router.get("/other").handler(ctx -> {
        assertEquals(true, ctx.request().params().isEmpty());
        ctx.response().end("/other");
      });
```